Add route info migration guide (#8467)
flutter/website: src/release/breaking-changes/route-information-uri.md

---
title: Migration guide for `RouteInformation.location`
description: Deprecation of `RouteInformation.location` and its related APIs
---

## Summary

> The APIs around the usages of RouteInformation.location was deprecated in the favor of RouteInformation.uri.

Reviewer: Consider something like:

```suggestion
`RouteInformation.location` and related APIs were deprecated in the favor of `RouteInformation.uri`.
```

## Context

> To handle mobile deeplinks from different web domains, the [`RouteInformation`][] needs to contain the authority information where the deeplinks are originated. The `uri` field was added to `RouteInformation` that captures the entire deeplink informations. Since then, every route-related parameters passing through framework were converted to full [`Uri`][] format. This led to deprecation of  imcompatible APIs.

Reviewer: Consider something like:

```suggestion
The [`RouteInformation`][] needs the authority information to handle mobile deeplinks from different web domains. The `uri` field was added to `RouteInformation` that captures the entire deeplink information and route-related parameters were converted to the full [`Uri`][] format. This led to deprecation of incompatible APIs.
```

```

Code after migration:

```dart
final RouteInformation myRoute = RouteInformation(uri: Uri.parse('/myroute'));
```

Code before migration:

```dart
final String myPath = myRoute.location;
```

Code after migration:

```dart
final String myPath = myRoute.uri.path;
```

Code before migration:

```dart
class MyObserverState extends State<MyWidget> with WidgetsBindingObserver {
  @override
  Future<bool> didPushRoute(String route) => _handleRoute(route);
}
```

Code after migration:

```dart
class MyObserverState extends State<MyWidget> with WidgetsBindingObserver {
  @override
  Future<bool> didPushRouteInformation(RouteInformation routeInformation) => _handleRoute(
    Uri.decodeComponent(
      Uri(
        path: uri.path.isEmpty ? '/' : uri.path,
        queryParameters: uri.queryParametersAll.isEmpty ? null : uri.queryParametersAll,
        fragment: uri.fragment.isEmpty ? null : uri.fragment,
      ).toString(),
    )
  );
}
```

Code before migration:

```dart
SystemNavigator.routeInformationUpdated(location: '/myLocation');
```

Code after migration:

```dart
SystemNavigator.routeInformationUpdated(uri: Uri.parse('/myLocation'));

> ```
> 
> 
> ## Timeline

Reviewer: ````suggestion
```

## Timeline
````

Please ignore if this was intentional :)
